app/models/concerns/date_produced.rb

```diff
@@ -0,0 +1,31 @@
+module DateProduced
+  extend ActiveSupport::Concern
+
+  included do
+    validate :date_produced_is_a_date
+  end
+
+  def date_produced_is_a_date
+    return if date_produced.blank?
+    errors.add(:date_produced, "should be a date in #{self.class.date_produced_placeholder}") unless date_produced.is_a?(Time)
```

Comment:
@devduarte @ftarulla Maybe the validator could allow the String form? Something like:

```ruby
case date_produced?
when Time
  # ok
when String
  # error unless matches regexp (or that it parses as a date)
else
  # error
end
```

I think it would permit to continue using the `#date_produced` accessor instead of the ivar?